test/token-service/test_tokenUpdateTransaction.js

```diff
@@ -0,0 +1,4557 @@
+        const response = await JSONRPCRequest("updateToken", {
+          tokenId: mutableTokenId,
+          autoRenewPeriod: "2591999",
+          commonTransactionParams: {
+            signers: [mutableTokenKey],
+          },
+        });
+        if (response.status === "NOT_IMPLEMENTED") this.skip();
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#13) Updates a mutable token with an auto renew period set to 8,000,001 seconds", async function () {
+      const autoRenewPeriod = "8000001";
+      const response = await JSONRPCRequest("updateToken", {
+        tokenId: mutableTokenId,
+        autoRenewPeriod: autoRenewPeriod,
+        commonTransactionParams: {
+          signers: [mutableTokenKey],
+        },
+      });
+
+      await retryOnError(async () => {
+        verifyTokenAutoRenewPeriodUpdate(mutableTokenId, autoRenewPeriod);
+      });
+    });
+
+    it("(#14) Updates a mutable token with an auto renew period set to 8,000,002 seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("updateToken", {
+          tokenId: mutableTokenId,
+          autoRenewPeriod: "8000002",
+          commonTransactionParams: {
+            signers: [mutableTokenKey],
+          },
+        });
+        if (response.status === "NOT_IMPLEMENTED") this.skip();
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+  });
+
+  describe("Expiration Time", function () {
+    async function verifyTokenExpirationTimeUpdate(tokenId, expirationTime) {
+      const parsedExpirationTime = Timestamp.fromDate(
+        new Date(Number(expirationTime) * 1000),
+      );
+
+      expect(parsedExpirationTime).to.deep.equal(
+        await (
+          await consensusInfoClient.getTokenInfo(tokenId)
+        ).expirationTime,
+      );
+
+      const mirrorNodeExpirationDateNanoseconds = await (
+        await mirrorNodeClient.getTokenData(tokenId)
+      ).expiry_timestamp;
+
+      // Convert nanoseconds got back from to timestamp
+      const mirrorNodeTimestamp = Timestamp.fromDate(
+        new Date(mirrorNodeExpirationDateNanoseconds / 1000000),
+      );
+
+      expect(parsedExpirationTime).to.deep.equal(mirrorNodeTimestamp);
+    }
+
+    //it("(#1) Updates an immutable token with a valid expiration time", async function () {
+    //  try {
+    //    const response = await JSONRPCRequest("updateToken", {
+    //      tokenId: immutableTokenId,
+    //      expirationTime: parseInt((Date.now() / 1000) + 5184000)
+    //    });
+    //    if (response.status === "NOT_IMPLEMENTED") this.skip();
+    //  } catch (err) {
+    //    assert.equal(err.data.status, "TOKEN_IS_IMMUTABLE");
+    //    return;
+    //  }
+    //
+    //  assert.fail("Should throw an error");
+    //});
```

Comment:
This was left commented out, and I haven't modified it. However, upon testing it now, it seems it may have been commented because it performs updates on the `expirationTime` prop and involves an immutableToken, which could indicate an error in the services.

Reply:
This was left commented out, and I haven't modified it.  However, after testing it now, it appears it may have been commented because it throws an invalid expiration time error, even though it is within the acceptable range.

Reply:
Maybe we should uncomment this and just `.skip` it?